Scripts/Source/minai_Sex.psc

```diff
@@ -60,29 +60,45 @@ bool Function CanAnimate(actor akTarget, actor akSpeaker)
 EndFunction
 
 Function Start1pSex(actor akSpeaker)
-  if bHasOstim && minai_UseOStim.GetValue() == 1.0
-    OThread.QuickStart(OActorUtil.ToArray(akSpeaker))
-  else
-    slf.Quickstart(akSpeaker)
+  if CanAnimate(akSpeaker, akSpeaker)
+    if bHasOstim && minai_UseOStim.GetValue() == 1.0
+      OThread.QuickStart(OActorUtil.ToArray(akSpeaker))
+    else
+      slf.Quickstart(akSpeaker)
+	EndIf
   EndIf
 EndFunction
 
 
 Function Start2pSex(actor akSpeaker, actor akTarget, actor Player, bool bPlayerInScene, string tags="")
-  if bHasOstim && minai_UseOStim.GetValue() == 1.0
-    ; Convert sexlab tags to ostim tags
-    if tags == "Anal"
-      tags = "analsex"
-    elseif tags == "vaginal"
-      tags = "vaginalsex"
-    elseif tags == "oral"
-      tags = "blowjob"
-    EndIf
-	Actor[] ostimActors = new Actor[2]
-	if bPlayerInScene
-	  ostimActors = OActorUtil.ToArray(Player, akSpeaker) as Actor[]
-	else
-	  ostimActors = OActorUtil.ToArray(akTarget, akSpeaker) as Actor[]
+  if CanAnimate(akTarget, akSpeaker)
+    if bHasOstim && minai_UseOStim.GetValue() == 1.0
+      ; Convert sexlab tags to ostim tags
+      if tags == "anal"
```

Comment:
Since you're referencing this in two places, let's add a function for this instead of repeating it

Reply:
(The logic to convert tags I mean)